File: 백준/Silver/11399. ATM/ATM.java
import java.io.*;
import java.util.*;

public class Main {

	public static void main(String[] args) throws Exception{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		int N = Integer.parseInt(br.readLine());
		int[] arr = new int[N];

		StringTokenizer st = new StringTokenizer(br.readLine());
		for (int i=0; i<N; i++) {
			arr[i] = Integer.parseInt(st.nextToken());
		}

		Arrays.sort(arr);
		int answer = arr[0];
		int nextNum = arr[0];

		for (int i=1; i<N; i++) {
			nextNum += arr[i];
			answer += nextNum;
		}

		System.out.println(answer);

	}

}

File: 백준/Silver/11399. ATM/ATM.kt
import java.io.*
import java.util.*

fun main() {
    /*
     * line1 N: 사람 수
     * line2 Pi: 인출 시간
     **/

    val br = BufferedReader(InputStreamReader(System.`in`))
    val N = br.readLine().toInt()
    val arr = br.readLine().split(" ").map{ it.toInt() }.sorted()

    var res = 0
    var sum = 0
    for (i in arr.indices) {
        sum += arr[i]
        res += sum
    }

    print(res)
}

